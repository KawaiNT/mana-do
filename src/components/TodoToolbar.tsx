import React from 'react';
import { isTodoCompleted } from '../utils';
import { TodoStatus } from '../models/todo';
import { TodoToolbarInterface } from '../models/todo';

const TodoToolbar = (props: TodoToolbarInterface) => {
  const activeTodos = props.todos.reduce(function (accum, todo) {
    return isTodoCompleted(todo) ? accum : accum + 1;
  }, 0);

  return (
    <div className='Todo__toolbar' data-testid="todo-toolbar">
      {props.todos.length > 0 ? (
        <input
          type='checkbox'
          checked={activeTodos === 0}
          onChange={props.onToggleAllTodo}
        />
      ) : (
        <div />
      )}
      <div className='Todo__tabs'>
        <button className={ props.showing === 'ALL' ? 'Action__btn' : ''} onClick={() => props.setShowing('ALL')}>
          All
        </button>
        <button
          className={ props.showing === 'ACTIVE' ? 'Action__btn' : ''}
          onClick={() => props.setShowing(TodoStatus.ACTIVE)}
        >
          Active
        </button>
        <button
          className={ props.showing === 'COMPLETED' ? 'Action__btn' : ''}
          onClick={() => props.setShowing(TodoStatus.COMPLETED)}
        >
          Completed
        </button>
      </div>
      <button onClick={props.onDeleteAllTodo}>
        Clear all todos
      </button>
    </div>
  );
};

export default TodoToolbar;
