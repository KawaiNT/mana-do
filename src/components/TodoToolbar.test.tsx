import React from 'react';
import { render } from '@testing-library/react';
import TodoToolbar from './TodoToolbar'

describe('TodoToolbar component test', () => {
  test('TodoToolbar render', () => {
    const setShowing = jest.fn()
    const onDeleteAllTodo = jest.fn()
    const onToggleAllTodo = jest.fn()
    const { asFragment } = render(<TodoToolbar todos={[]} onDeleteAllTodo={onDeleteAllTodo} setShowing={setShowing} onToggleAllTodo={onToggleAllTodo} />)

    expect(asFragment(<TodoToolbar todos={[]} onDeleteAllTodo={onDeleteAllTodo} setShowing={setShowing} onToggleAllTodo={onToggleAllTodo} />)).toMatchSnapshot()
  })

  test('TodoToolbar should be in the document', () => {
    const setShowing = jest.fn()
    const onDeleteAllTodo = jest.fn()
    const onToggleAllTodo = jest.fn()
    const { getByTestId } = render(<TodoToolbar todos={[]} onDeleteAllTodo={onDeleteAllTodo} setShowing={setShowing} onToggleAllTodo={onToggleAllTodo} />)
    const todoToolbar = getByTestId('todo-toolbar')

    expect(todoToolbar).toBeInTheDocument()
  })
})