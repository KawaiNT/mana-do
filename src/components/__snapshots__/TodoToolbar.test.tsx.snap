// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoToolbar component test TodoToolbar render 1`] = `
<DocumentFragment>
  <div
    class="Todo__toolbar"
    data-testid="todo-toolbar"
  >
    <div />
    <div
      class="Todo__tabs"
    >
      <button
        class=""
      >
        All
      </button>
      <button
        class=""
      >
        Active
      </button>
      <button
        class=""
      >
        Completed
      </button>
    </div>
    <button>
      Clear all todos
    </button>
  </div>
</DocumentFragment>
`;
